Ports/RISC-V/RV32/GCC/os_cpu_a.S

```diff
@@ -139,9 +138,43 @@ OSStartHighRdy:
     lw     t5,  29 * 4(sp)
     lw     t6,  30 * 4(sp)
 
+    #Restore f0 to f31 registers (FP registers)
+    flw    f0,  32 * 4(sp)
+    flw    f1,  33 * 4(sp)
+    flw    f2,  34 * 4(sp)
+    flw    f3,  35 * 4(sp)
+    flw    f4,  36 * 4(sp)
+    flw    f5,  37 * 4(sp)
+    flw    f6,  38 * 4(sp)
+    flw    f7,  39 * 4(sp)
+    flw    f8,  40 * 4(sp)
+    flw    f9,  41 * 4(sp)
+    flw    f10, 42 * 4(sp)
+    flw    f11, 43 * 4(sp)
+    flw    f12, 44 * 4(sp)
+    flw    f13, 45 * 4(sp)
+    flw    f14, 46 * 4(sp)
+    flw    f15, 47 * 4(sp)
+    flw    f16, 48 * 4(sp)
+    flw    f17, 49 * 4(sp)
+    flw    f18, 50 * 4(sp)
+    flw    f19, 51 * 4(sp)
+    flw    f20, 52 * 4(sp)
+    flw    f21, 53 * 4(sp)
+    flw    f22, 54 * 4(sp)
+    flw    f23, 55 * 4(sp)
+    flw    f24, 56 * 4(sp)
+    flw    f25, 57 * 4(sp)
+    flw    f26, 58 * 4(sp)
+    flw    f27, 59 * 4(sp)
+    flw    f28, 60 * 4(sp)
+    flw    f29, 61 * 4(sp)
+    flw    f30, 62 * 4(sp)
+    flw    f31, 63 * 4(sp)
+
 # Compensate for the stack pointer
-    # sp = sp + 32 * 4
-    addi   sp, sp, 32 * 4
+    # sp = sp + 64 * 4
```

Comment:
For storing 0 to 30 registers we used 32. hence my previous comment why we are not starting from 31. for FP. So just be careful here. Kindly check again. So either
1. Change this to 65
2. Change index to 31 instead of 32 to start saving FP registers
3. Or review and confirm we had a hole in 31st index earlier. It was just left unused to make it a multiple of 32 register saves.

Reply:
31 stores mepc, we need to treat this special, before its need a temp register. since we also need to save restore this temp register its need to be done carefully in sequence.